DFBUGS-3771: [release-4.19] Remove csi toleration specification & Reserve the rook-ceph-operator-config configmap for overrides

malayparida2000 opened the pull request:
Manual Cherry-pick of https://github.com/red-hat-storage/ocs-operator/pull/3422
Till 4.18, tolerations for csi pods were being added to the storageCluster CR, and those were being passed to rook ceph operator via the rook-ceph-operator-config configMap. From 4.19 onwards, rook no longer deploys the csi, it's now done by ceph-csi-operator.
To add tolerations to csi pods user has to now directly edit the csi driver CR, ref-https://issues.redhat.com/browse/DFBUGS-2764. So to avoid confusion, the code relating to adding the csi tolerations via ocs-operator should be removed.

Then We can just create the configmap and not reconcile it.

malayparida2000 committed:
Add generated changes

Signed-off-by: Malay Kumar Parida <mparida@redhat.com>

malayparida2000 commented:
/jira refresh

malayparida2000 commented:
/jira refresh

malayparida2000 commented:
/jira refresh

Madhu-1 commented:
@malayparida2000 i think this contains API changes. do we do API changes in the patch release?

malayparida2000 commented:
> i think this contains API changes. do we do API changes in the patch release?

We do not have any strict standard about it, But we normally avoid API changes in patch releases. But if you notice the actual changes the fields that are removed are status fields. And status fields are always set by the owner operator(i.e us), So I don't forsee any risk with this. @nb-ohad WDYT?

Madhu-1 commented:
> > i think this contains API changes. do we do API changes in the patch release?
> 
> We do not have any strict standard about it, But we normally avoid API changes in patch releases. But if you notice the actual changes the fields that are removed are status fields. And status fields are always set by the owner operator(i.e us), So I don't forsee any risk with this. @nb-ohad WDYT?

okay, i thought this is spec changes, i hope no customer build any CI tools for it (any way it wont work if its set here)

nb-ohad commented:
> > i think this contains API changes. do we do API changes in the patch release?
> 
> We do not have any strict standard about it, But we normally avoid API changes in patch releases. But if you notice the actual changes the fields that are removed are status fields. And status fields are always set by the owner operator(i.e us), So I don't forsee any risk with this. @nb-ohad WDYT?

We should not be removing in a z stream. The reason is that we don't know what customer are expecting. They might have automations that relay on these status fields. The proper way to do it is to mark them as deprecated on the next y release then remove them completely in the y release the follows

malayparida2000 committed:
Remove csi toleration specification & Reserve configmap for overrides

Till 4.18, tolerations for csi pods were being added to the
storageCluster CR, and those were being passed to rook ceph operator
via the rook-ceph-operator-config configMap. From 4.19 onwards, rook no
longer deploys the csi, it's now done by ceph-csi-operator.
To add tolerations to csi pods user has to now directly edit the csi
driver CR, ref-https://issues.redhat.com/browse/DFBUGS-2764. So to avoid
confusion, the code relating to adding the csi tolerations via
ocs-operator should be removed.

Then We can just create the configmap and not reconcile it.

Signed-off-by: Malay Kumar Parida <mparida@redhat.com>
